enumerables.rb

```diff
@@ -0,0 +1,134 @@
+module Enumerable
+  # My Each Method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
+
+  # My Each With Index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter], counter)
+      counter += 1
+    end
+    arr
+  end
+
+  # My Select
+  def my_select(arr)
+    new_array = []
+    arr.my_each do |index_value|
+      new_array << index_value if yield(index_value)
+    end
+    new_array
+  end
+
+  # My All?
+  def my_all?(arr)
+    result = false
+    arr.my_each do |item|
+      result = true if yield(item == true)
+    end
+    result
+  end
+
+  # My Any?
+  def my_any?(arr)
+    result = false
+    arr.my_each do |item|
+      next unless yield(item)
+
+      result = true
+    end
+    result
+  end
+
+  # My None?
+  def my_none?(arr)
+    result = true
+    arr.my_each do |item|
+      result = false if yield(item)
+    end
+    result
+  end
```

Comment:
## Optional argument
- [ ] Add  optional argument to this method. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)


```ruby
def my_none?(argument = nil)
```

## Hashes, range and array self
- [ ] Test with hashes and ranges, then compare with original `none` Ruby method
- [ ] Fix your method to work with the tests. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)

**Example:**

`test_file.rb`
```ruby
(0..9).my_none{
# or
Range.new(0, 9)

# HASHES
{ a: 1, b: 2, c: 3, d: 4, e: 5 }
```